crates/sargon/src/system/sargon_os/sargon_os_accounts.rs

```diff
@@ -292,6 +292,76 @@ impl SargonOS {
 
         Ok(())
     }
+
+    /// Updates the accounts `updated` by mutating current profile and persisting
+    /// the change to secure storage. Throws `UnknownAccount` error if some account
+    /// is not found.
+    ///
+    /// # Emits Event
+    /// Emits `Event::ProfileModified { change: EventProfileModified::AccountsUpdated { addresses } }`
+    pub async fn update_accounts(&self, updated: Accounts) -> Result<()> {
+        self.update_profile_with(|p| {
+            let mut result = Ok(());
+            for updated_account in updated.clone() {
+                if p.update_account(&updated_account.address, |old| {
+                    *old = updated_account.clone()
+                })
+                .is_none()
+                {
+                    result = Err(CommonError::UnknownAccount);
+                    break;
+                }
+            }
```

Comment:
if an array with 5 account has only the third one as missing, this would update the first 2 and return an error. Wouldn't that leave the Profile in a corrupted state?

Reply:
hmm better than nothing I think? The more the merrier so to speak. Better mark `N` of `M` tombstoned accounts as such, even if `N < M`.

Reply:
hmm better than nothing I think? The more the merrier so to speak. Better mark `N` of `M` tombstoned accounts as such, even if `N < M`.

Reply:
for this case yeah, but as a general rule (this function is to update accounts, not only to tombstone them) not sure if it is better partially updating.

Reply:
I would opt for having a consistency check done before calling update_profile. 
Additionally what would be a scenario when we can end up calling `update_accounts` with a missing account? If there is a scenario, should we throw an error, or doing an update on a non-existing account can be considered noop?

Reply:
We SHOULD know if we should use ADD or UPDATE, that is really something we should keep track of. So I think `update` ought to always be failing for unknown entities. It is a programmer error.

Reply:
>  if the account is missing.
sounds like @GhenadieVP you thought I meant to fail if parameter `updated: Accounts` missed some of the existing accounts. That is not what I meant. That would be strange!

I'm saying if `updated: Accounts` contains a **new** account, which is not in profile, we ought to fail.

And I'm also saying that I think we should revert the whole operation if an update of any account fails. Which feels more prudent and **easy to reason about**.